server/routerlicious/packages/services-shared/src/socketIoServer.ts

```diff
@@ -121,13 +109,21 @@ export function create(
             socketIoAdapterConfig?.shouldDisableDefaultNamespace);
 
         adapter = redisSocketIoAdapter.RedisSocketIoAdapter as any;
-    }
-    else {
-        adapter = socketIoRedis({ pubClient: pub, subClient: sub });
+    } else {
+        adapter = createAdapter(pub, sub);
     }
 
-    io.attach(server);
-    io.adapter(adapter);
+    // Create and register a socket.io connection on the server
+    const io = new Server(server, {
```

Comment:
For reference, here are the options I use in pushchannel:
```typescript
			allowEIO3: true, // enable socket.io v2 clients
			adapter: RedisSocketIoAdapter as any,
			perMessageDeflate: false,
			cookie: false,
			serveClient: false,
			transports: [ "websocket" ],
			pingInterval: 30 * 1000,
			maxHttpBufferSize: 1e6, // 1mb
			// same cors options as fastify-cors
			cors: {
				origin: true,
				credentials: true
			}
```
I would recommend setting transports to websocket to ensure long polling is never used. And maybe check if the other options are applicable for you as well.

Reply:
Took cors and transports from yours. The rest were just the defaults so I let them be (except pingInterval, which is 25s by default compared to your 30s).